docs/design/2021-08-23-history-hot-regions.md

```diff
@@ -0,0 +1,269 @@
+# Record history hotspot information in TiDB_HOT_REGIONS_HISTORY
+
+- Author(s): [@qidi1](https://github.com/qidi1), [@icepigzdb](http://github.com/icepigzdb)
+- Discussion PR: https://github.com/pingcap/tidb/pull/27487
+- Tracking Issue: https://github.com/pingcap/tidb/issues/25281
+
+## Table of Contents
+
+  * [Introduction](#introduction)
+  * [Motivation or Background](#motivation-or-background)
+  * [Detailed Design](#detailed-design)
+    + [Existing TIDB HOT REGIONS](#existing-tidb-hot-regions)
+    + [Table Header Design](#table-header-design)
+    + [PD](#pd)
+    + [TiDB](#tidb)
+  * [Test Design](#test-design)
+    + [Functional Tests](#functional-tests)
+    + [Scenario Tests](#scenario-tests)
+    + [Compatibility Tests](#compatibility-tests)
+    + [Benchmark Tests](#benchmark-tests)
+  * [Impacts & Risks](#impacts--risks)
+  * [Investigation & Alternatives](#investigation--alternatives)
+
+## Introduction
+
+Create a new table `TIDB_HOT_REGIONS_HISTORY` in the `INFORMATION_SCHEMAT` schema to retrieve history hotspot regions stored by PD periodically.  
+
+## Motivation or Background
+
+TiDB has a memory table `TIDB_HOT_REGIONS` that provides information about hotspot regions. But it only shows the current hotspot information. This leads to the fact that when DBAs want to query historical hotspot details, they have no way to find the corresponding hotspot information.
+
+According to the [documentation](https://docs.pingcap.com/tidb/stable/information-schema-tidb-hot-regions) for the current `TIDB_HOT_REGIONS` table,  it can only provides information about recent hotspot regions calculated by PD according to the heartbeat from tikv.  It is inconvenient to obtain hotspot regions of past time, and locate which store the region is. For ease of use, we can store extened hotspot region infomation in PD. The DBA can query hotspot regions within a specified period by such statement in TiDB:
+
+```SQL
+SElECT * FROM information_schema.tidb_hot_regions_history WHERE update_time>='2019-04-01 00:00:00' and update_time<='2019-04-01 00:01:00';
```

Comment:
add some other user case, such as group by store_id.

Reply:
add in new comment